app/warrior/knight.py

```diff
@@ -0,0 +1,36 @@
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list | None,
+            weapon: dict | None,
+            potion: dict | None
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = 0
+
+    def apply_armour(self) -> None:
+        for arm in self.armour:
+            self.protection += arm["protection"]
```

Comment:
The `apply_armour` method assumes that `self.armour` is always a list. If `self.armour` is `None`, this will raise a `TypeError`. Consider adding a check to ensure `self.armour` is not `None` before iterating over it.